scripts/validation/embedded-examples-validation.js

```diff
@@ -0,0 +1,119 @@
+const fs = require('fs');
+const yaml = require('js-yaml');
+const { Parser } = require('@asyncapi/parser');
+const mergePatch = require('json-merge-patch');
+const jsonpointer = require('jsonpointer');
+const parser = new Parser();
+
+// Read the markdown file
+const markdownContent = fs.readFileSync('../../spec/asyncapi.md', 'utf8');
+
+// Function to extract comments and examples from the markdown content
+function extractCommentsAndExamples(content) {
+  const combinedRegex = /<!--\s*asyncapi-example-tester:\s*({.*?})\s*-->\s*\n```(.*)?\n([\s\S]*?)\n```/g;
+  let match;
+  const combinedData = [];
+
+  while ((match = combinedRegex.exec(content)) !== null) {
+    try {
+      const json = JSON.parse(match[1]);
+      const format = match[2].trim();
+      const exampleContent = match[3].trim();
+
+      let example;
+      if (format === 'json') {
+        example = JSON.parse(exampleContent);
+      } else if (format === 'yaml') {
+        example = yaml.load(exampleContent);
+      } else {
+        throw new Error(`Unsupported format: ${format}`);
+      }
+
+      combinedData.push({
+        name: json.name,
+        json_path: json.json_path,
+        example: example,
+        format: format,
+      });
+    } catch (e) {
+      console.error("Failed to parse comment JSON or example:", match[1], e);
+      // process.exit(1);
+    }
+  }
+
+  return combinedData;
+}
+
+// Extract comments and examples from the markdown file
+const combinedData = extractCommentsAndExamples(markdownContent);
+
+// Function to apply JSON Merge Patch updates to the document
+function applyUpdates(updates, baseDoc) {
+  updates.forEach(update => {
+    try {
+      if (update.json_path === "$") {
```

Comment:
`$` is not a valid json pointer right? I guess we would need to just change it by having an empty json_path (actually `json_pointer`) field

Reply:
Additionally, you can remove this check and directly run the `else` part, because most probably `jsonpointer.get(baseDoc, '');` will return a pointer to the root. Just try it

Reply:
Let me see that...

Reply:
@smoya I tried what you said but was getting this error while attempting it:
```bash
Error during validations: TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'Object'
    --- property 'undefined' closes the circle
    at JSON.stringify (<anonymous>)
    at logDocument (/home/animeshkumar923/open-source/asyncapi-spec/scripts/validation/test-v1.js:92:35)
    at /home/animeshkumar923/open-source/asyncapi-spec/scripts/validation/test-v1.js:115:3
    at Array.map (<anonymous>)
    at Object.<anonymous> (/home/animeshkumar923/open-source/asyncapi-spec/scripts/validation/test-v1.js:108:41)
    at Module._compile (node:internal/modules/cjs/loader:1376:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1435:10)
    at Module.load (node:internal/modules/cjs/loader:1207:32)
    at Module._load (node:internal/modules/cjs/loader:1023:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
```
I don't know exactly what's going wrong so I'm going with the previous approach. For that `if-else` approach, it adds correctly the values of root doc in the updated doc.

Reply:
But my point still is valid: `$` I not a valid json pointer

Reply:
> But my point still is valid: `$` I not a valid json pointer

Yes, that's correct. I've changed it to point to `if (update.json_path === "")`